compiler-builtins/src/mem/x86_64.rs

```diff
@@ -22,13 +22,12 @@ use core::{intrinsics, mem};
 #[inline(always)]
 #[cfg(target_feature = "ermsb")]
 pub unsafe fn copy_forward(dest: *mut u8, src: *const u8, count: usize) {
-    // FIXME: Use the Intel syntax once we drop LLVM 9 support on rust-lang/rust.
-    core::arch::asm!(
-        "repe movsb (%rsi), (%rdi)",
+    asm!(
+        "repe movsb [rdi], rsi",
```

Comment:
In the original asm both rsi and rdi are dereferenced, here it’s only rdi. Is this still equivalent for some reason?

Reply:
I don't think it is. I fixed this.